rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -107,40 +107,36 @@ std::shared_ptr<PublisherData> PublisherData::make(
     return nullptr;
   }
 
-  zenoh::ZResult result;
-  std::optional<zenoh::ext::PublicationCache> pub_cache;
-  zenoh::KeyExpr pub_ke(entity->topic_info()->topic_keyexpr_);
+  auto adv_pub_opts = zenoh::ext::SessionExt::AdvancedPublisherOptions::create_default();
+  adv_pub_opts.publisher_detection = true;
+  adv_pub_opts.sample_miss_detection = true;
```

Comment:
Looking at the [documentation for this option](https://zenoh-cpp.readthedocs.io/en/latest/ext.html#_CPPv4N5zenoh3ext10SessionExt24AdvancedPublisherOptions21sample_miss_detectionE), it sounds like this should be set `true` only if `adapted_qos_profile.reliability == RMW_QOS_POLICY_RELIABILITY_RELIABLE`? Could you clarify how this differs from setting `pub_opts.reliability = Z_RELIABILITY_RELIABLE`?

Reply:
The reliability option can still be set by the normal publisher option. See [here](https://github.com/ZettaScaleLabs/rmw_zenoh/blob/9cc776bd8cc7fddfea8dec0ad703a22d3a6c96d3/rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp?plain=1#L128). The reliability in zenoh is based on TCP reliability per see. The sample_miss_detection here is a custom data retransmission feature cached on the publisher side in the sense of transient_local duriability.